components/SelectOptionDetailed.tsx

```diff
@@ -0,0 +1,65 @@
+export enum Flag {
+  Danger,
+  Warning,
+  OK,
+}
+
+type OptionText = {
+  text: string;
+  flag?: Flag;
+};
+
+type OptionDetails = {
+  [label: string]: OptionText;
+};
+
+const setFlagColorText = (flag?: Flag) => {
+  switch (flag) {
```

Comment:
You may create a map to set the colors. So typing would generate an error if the dev doesn't set a color when adding a new flag.

```
const flagColorClassMap: {
  [flag in Flag]: string;
} = {
  [Flag.Danger]: 'text-red',
  [Flag.Warning]: 'text-orange',
  [Flag.OK]: 'text-green',
};

const setFlagColorText = (flag?: Flag) => {
  if (!flag) {
    return '';
  }

  return flagColorClassMap[flag];
};
```